pom.xml

```diff
@@ -1143,6 +1149,32 @@
         </executions>
       </plugin>
 
+      <!-- Use commit time for reproducible builds - https://maven.apache.org/guides/mini/guide-reproducible-builds.html -->
+      <plugin>
+        <groupId>pl.project13.maven</groupId>
+        <artifactId>git-commit-id-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>get-the-git-infos</id>
+            <goals>
+              <goal>revision</goal>
+            </goals>
+            <phase>initialize</phase>
+          </execution>
+        </executions>
+        <configuration>
+          <generateGitPropertiesFile>true</generateGitPropertiesFile>
+          <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties</generateGitPropertiesFilename>
```

Comment:
if this generates a file, add it to .gitignore?

Reply:
files will already be ignored bc they're generated in `<module>/target/classes` build directories and jars